Embed marker icons in the html file.

frslm opened the pull request:
This allows HTML files with markers to be shared  (#73), since all required marker icons are now embedded in the HTML file itself.

Also has the added benefit of rendering markers on Windows machines (#3, #18, #57, #62, #63, and #97).

frslm commented:
Hi @vgm64, I've put up a few PRs that address several issues - let me know if this repo is still being maintained?

sandrotosi commented:
@frslm should we make this an conditional option? how much bigger is the HTML page with the markers in it?

frslm commented:
@sandrotosi each embedded marker icon will make the file ~0.7 KB larger. (Note that if I render 100 markers with the same icon, it'll still count as only 1 embedded icon, so the increase in size will still be ~0.7 KB, with the total file size being ~20-30 KB.)

If this increase in size is a concern, then we could add a memory-saving option that doesn't embed icons, but that would produce HTML files that can't be shared (since they'll need the `gmplot/markers` directory to pull the icons).

Personally, I don't think this increase in size is an issue, especially since this change also gets rid of all the duplicate `google.maps.MarkerImage()` lines (in a 100-marker file, for example, this'll actually save ~10 KB worth of lines, which more than makes up for the ~0.7 KB increase from one embedded icon).

What do you think?

frslm commented:
Merged since this has been open for too long, and fixes many issues. If there are any unaddressed concerns, please bring them up in a new issue and we can tackle them there.
